app/services/authn_context_resolver.rb

```diff
@@ -100,4 +146,8 @@ def acr_ial_component_values
       component_value.name.include?('ial') || component_value.name.include?('loa')
     end
   end
+
+  def sp_in_biometric_pilot?
+    @service_provider.biometric_ial_allowed?
```

Comment:
i don't *think* we currently need this check, since the feature flag gate is before this. (although if/when i get my "ignore unknown authncontext/acr_values" in, that would change, but that's a future issue)

Reply:
Yeah, I would expect this logic to be handled when we do authn context authorization (in the OIDC form and SAML request validated). We do seem to have logic in there for validating that the SP can request biometric. We also have similar logic for IALMax and identity proofing. This looks like it is mixing concerns about authorizing authn context with resolving the authn context.

Related to authorizing authn contexts, I did have the thought that we could simplify our authn context authorization logic a bit if we moved that concern into the authn context itself e.g.

```
IAL2_BIO_REQUIRED = ComponentValue.new(
  name: Saml::Idp::Constants::IAL2_BIO_REQUIRED_AUTHN_CONTEXT_CLASSREF,
  description: 'IAL2 - identity proofing with biometric comparison (rev3)',
  implied_component_values: [],
  requirements: [:aal2, :identity_proofing, :biometric_comparison, :two_pieces_of_fair_evidence],
  sp_authorization: ->(sp) { sp.biometric_ial_allowed? }
).freeze
```

Then if you find the authn context reference object in the form you can call it with the SP to authorize it. That would let us remove a bunch of bespoke authn context authorization.

Reply:
And, to be clear, I'm not suggesting that authn context authorization needs to be completely re-built in this pull-request. We should do what works and come back around if we like the idea.

Reply:
oh that's an interesting idea, so the component itself is holding that information